### DISCUSSION TITLE
How to zoom by code?

### DISCUSSION THREAD

**Discussion by GF-Huang** [Live-Charts/LiveCharts2#257] [Q&A] [answered]
My chart's X-axis data grow in real time, but the chart not display the newer data, I need to scroll down my mouse wheel to do it.

**Comment by ghost** [comment, marked as answer]
Did you check [Lines - AutoUpdate](https://github.com/beto-rodriguez/LiveCharts2/blob/master/samples/ViewModelsSamples/Lines/AutoUpdate/ViewModel.cs) or [General - MultiThreading](https://github.com/beto-rodriguez/LiveCharts2/blob/master/samples/ViewModelsSamples/General/MultiThreading/ViewModel.cs) in the Live Chart Samples?

**Reply by GF-Huang** [reply]
There is nothing special about these two examples. I had use `ObservableCollection`, but I still need to zoom the chart to see newer data manually.

**Reply by ghost** [reply]
Are you using WinForms? see #253

**Reply by GF-Huang** [reply]
WPF.

**Reply by ghost** [reply]
In my WPF project the sample code works fine. Please share some code showing the issue.

**Reply by GF-Huang** [reply]
Sorry I can't share my full project. I use `Fody.PropertChanged` in my project, so I don't need to do the notify property changed manually.

My ViewModel:

![image](https://user-images.githubusercontent.com/4510984/138888481-244b3749-cde8-4ef2-8e7c-6f0b4c63e910.png)

`OptimizedObservableCollection` just add some `Range` methods.

![image](https://user-images.githubusercontent.com/4510984/138888540-abb0c482-b598-4676-b04a-bcf5bb608206.png)

**Reply by GF-Huang** [reply]
I think you're right, before I manually zoom the chart, it will auto zoom to the newer data; but after I manually zoom, it will not auto zoom, how can I set it back to the initial status for auto zooming?

**Reply by beto-rodriguez** [reply]
Set the limits of your axis back to `null`

https://github.com/beto-rodriguez/LiveCharts2/blob/master/docs/components/1.7.axes.md#maxlimit-and-minlimit-properties
